policyengine_us/variables/gov/states/nj/tax/income/credits/eitc/nj_childless_eitc_age_eligible.py

```diff
@@ -0,0 +1,37 @@
+from policyengine_us.model_api import *
+
+
+class nj_childless_eitc_age_eligible(Variable):
+    value_type = bool
+    entity = TaxUnit
+    label = "New Jersey Eligible for EITC"
+    definition_period = YEAR
+    reference = "https://law.justia.com/codes/new-jersey/2022/title-54a/section-54a-4-7/"
+    defined_for = StateCode.NJ
+
+    def formula(tax_unit, period, parameters):
+        # Return True if all federal EITC conditions are met, except with modified age paramaters and household has no children.
+
+        # Check if filing status is separate.
+        filing_status = tax_unit("filing_status", period)
+        separate = filing_status == filing_status.possible_values.SEPARATE
+
+        # Check if tax unit has children.
+        no_children = tax_unit("tax_unit_children", period) == 0
+
+        # Get the NJ EITC paramaeter tree.
+        p = parameters(period).gov.states.nj.tax.income.credits.eitc
+
+        # Check if the filer meets NJ EITC age requirements.
+        person = tax_unit.members
+        age = person("age", period)
+        min_age = p.eligibility.age.min
+        max_age = p.eligibility.age.max
+        meets_age_requirements = (age >= min_age) & (age <= max_age)
```

Comment:
Should this be for head or spouse? I could see some corner cases where a dependent meets the age requirement.

Reply:
Yea I think doing it for the head makes more sense. I tried to make it match the federal eitc eligible calculation but I think this (new pushed change) is a bit better. I suppose another edge case is if the head of household were under 18 but someone in the household were 18 or older.

Reply:
Sorry so if only the spouse is age-eligible it wouldn't count?

Reply:
Maybe I'm misunderstanding the legal code? It does seem a bit weird but neither the 1040 nor legal code (unless the spouse is implicitly included in people that can claim in the tax unit) mention the spouse for the case of a 18-25 year old without a child claiming the credit:

Legal code:
(4) A resident individual who is at least 18 years of age or older, but cannot claim a qualifying child as defined under section 152 of the federal Internal Revenue Code of 1986 (26 U.S.C. s.152) for the federal earned income tax credit, shall be eligible for the New Jersey earned income tax credit if that resident individual is ineligible to claim the federal earned income tax credit due to age requirements. The resident individual shall meet all qualifications, except for the minimum or maximum age, for the federal earned income tax credit in order to be eligible for the New Jersey earned income tax credit. Calculation of the New Jersey earned income tax credit available to individuals pursuant to this paragraph shall be predicated on the federal maximum amount for taxpayers with no qualifying child for each taxable year beginning on and after January 1, 2020.

From the worksheet: 
If you were at least 18 years old, you may be eligible for an
NJEITC even if you did not meet the age requirement for a
federal EIC. The maximum age limit has been eliminated.

Reply:
OK I think we can do what we do in the federal EITC then, which is any person in the tax unit:

https://github.com/PolicyEngine/policyengine-us/blob/d0c9fbc0da974c6c9241ffa40abc0ac7f4ec82ea/policyengine_us/variables/gov/irs/credits/earned_income/eitc_eligible.py#L38